datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -515,6 +521,23 @@ fn get_valid_types(
 
             vec![vec![valid_type; *number]]
         }
+        TypeSignature::Comparable(num) => {
+            function_length_check(current_types.len(), *num)?;
+            let mut target_type = current_types[0].to_owned();
+            for data_type in current_types.iter().skip(1) {
+                if let Some(dt) = comparison_coercion_numeric(&target_type, data_type) {
```

Comment:
It seems like `TypeSignature::Comparable`  has different rules that, for example, `=` or `<`  (it prefers numeric)

I think this is fine, but should probably be documented better / more explicitly as it might be expected

Also the coercion / defaulting to `Utf8` might also be a good thing to document